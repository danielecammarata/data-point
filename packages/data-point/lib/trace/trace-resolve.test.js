/* eslint-env jest */

const TraceResolve = require("./trace-resolve");

const NS_PER_SEC = 1e9;

const mockedTime = [1, NS_PER_SEC * 1];

describe("augmentTraceNodeDuration", () => {
  let mockGetDuration;
  afterAll(() => {
    mockGetDuration.mockRestore();
  });
  it("should set duration", () => {
    mockGetDuration = jest
      .spyOn(TraceResolve, "getDurationNs")
      .mockImplementation(t => {
        return `${t}&duration`;
      });
    const traceNode = {
      hrtime: "hrtime"
    };
    expect(TraceResolve.augmentTraceNodeDuration(traceNode)(true)).toEqual(
      true
    );
    expect(traceNode).toHaveProperty("durationNs", "hrtime&duration");
  });
});

describe("createTracedAccumulator", () => {
  let mockhrTime;
  afterAll(() => {
    mockhrTime.mockRestore();
  });
  it("should create new accumulator with traceNode appended", () => {
    mockhrTime = jest.spyOn(process, "hrtime").mockImplementation(() => {
      return mockedTime;
    });
    const accumulator = {};
    const result = TraceResolve.createTracedAccumulator(accumulator, "reducer");
    expect(mockhrTime).toBeCalled();
    expect(result).toMatchSnapshot();
    // should not mutate accumulator
    expect(accumulator).toEqual({});
  });
});

describe("augmentAccumulatorTrace", () => {
  let mockCreateTracedAccumulator;
  afterAll(() => {
    mockCreateTracedAccumulator.mockRestore();
  });
  it("should create new accumulator with traceNode appended", () => {
    mockCreateTracedAccumulator = jest
      .spyOn(TraceResolve, "createTracedAccumulator")
      .mockImplementation((input, reducer) => {
        return Object.assign({}, input, {
          traceNode: "traceNode",
          reducer
        });
      });
    const acc = {
      traceGraph: []
    };
    const result = TraceResolve.augmentAccumulatorTrace(acc, "reducer");
    expect(result).toMatchSnapshot();
  });
});
