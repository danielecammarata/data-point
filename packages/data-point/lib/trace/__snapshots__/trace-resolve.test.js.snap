// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`augmentAccumulatorTrace should create new accumulator with traceNode appended 1`] = `
Object {
  "reducer": "reducer",
  "traceGraph": Array [
    "traceNode",
  ],
  "traceNode": "traceNode",
}
`;

exports[`createTracedAccumulator should create new accumulator with traceNode appended 1`] = `
Object {
  "traceNode": Object {
    "hrtime": Array [
      1,
      1000000000,
    ],
    "id": "1",
    "parent": undefined,
    "reducer": "reducer",
    "timeStartNs": 2000000000,
  },
}
`;
